### ISSUE TITLE
In the level editor, the space between the option tiles and the level is dependent on the number of option tiles

### ISSUE THREAD

**Issue by maximyzer** [david-pfx/PuzzleScriptNext#3]
**Describe the bug**
In the level editor, the space between the top part (the tiles you can select to build your level) and the bottom part (the level itself) is not static. 

It is equal to the number of rows you have in the top part + 2.

**Reproduction Steps**
1. Have a bunch of possible tiles you can add in a level
2. Change the size of the level to be less wide than the number of tiles you can add
    Ex. If you have 5 tiles, make your level only 2 tiles wide
3. The less wide you level is, the more space between the tiles and the level will grow  

**What did you expect would happen?**
Only have 1 line between the tiles and the level

**OS & Browser version, desktop/mobile**
Windows 11, Chrome

**Screenshots, example projects, other relevant issues, etc.**
![image](https://github.com/david-pfx/PuzzleScriptNext/assets/13802168/d7c87a1c-438e-486e-abdb-f8bc13c238db)
![image](https://github.com/david-pfx/PuzzleScriptNext/assets/13802168/9d847890-d6a7-4620-b25a-6e16a0c8fadf)

**Comment by maximyzer** [comment]
@david-pfx I haven't tried editing the engine yet, but I'd be willing to check this one out

**Comment by maximyzer** [comment]
I have a fix, I'll make a pr later today !

**Comment by maximyzer** [comment]
@david-pfx Sorry to bother you with pings. 
I don't seem to have the rights to create branches on the origin to create PR to fix this. 

What's the procedure to contribute, giving rights to teh repo or do I have to create a fork and merge back from there ?

Thanks !

**Comment by david-pfx** [comment]
Ping away! 
The GitHub process is you have to create a fork, then create your branch on that, and then submit that.
Don't keep me in suspense. What did you find?

**Comment by maximyzer** [comment]
Good, will do that probably tomorrow, I’m sick in bed right now !

The glyphCount() function returns the list of every element in the legend section instead of only the glyphs with a single character that can be added in the levels.

Meaning the empty lines are actually glitched glyphs that have sone space drawn, buy are not rendered.

And I just want to say thanks for your effort on porting the new base stuff from puzzlescript into ps+, there’s really good stuff in there!

**Comment by david-pfx** [comment]
Is that a different bug? Doesn't sound like a prelude thing.

Glad you like it. The aim is to track PS and PS+, try to import bits of P:S or others where possible, and add my own ideas. 

Need anything? Just ask.

**Comment by maximyzer** [comment]
Yeah if you scroll up you can see this bug. I filed a new one this morning

**Comment by david-pfx** [comment]
On review, I think the behaviour you are seeing is 'by design'. The graphics layout (which you can find in `canvasResize`) goes to some trouble to ensure that the cell size and width are integral in size. In normal play the level grid is centered vertically, so there are gaps top and bottom. If you resize the window you will see the gaps follow, and sometimes there are none.

In the level editor the glyph display is placed close to the top, so there is a variable gap between glyphs and level. Again, resize the screen window to see it change.

In PS Next the fix you propose should have no effect. The function `GlyphCount()` is not used for this. The calculation of `editorRowCount` is based on `glyphImages.length` (line 977):

```
        editorRowCount = Math.ceil(glyphImages.length/(screenwidth-1));
```

Thanks for putting the effort into this.

**Comment by maximyzer** [comment]
I did my PR a bit too fast, I had tested it on a clone of your branch that was pulled before your last changes.
I ported it to my new fork, but the glyphCount() function wasn't used like before.
I will rework my PR.

Because I can't believe this is by design for 2 reasons.
1. It wasn't like that before
2. Without my fix the "scroll wheel option to cycle through the glyph" is broken, it continue scrolling through invalid options and you lose the white outline selector


----

This is the level editor of my game in PS Plus
![image](https://github.com/david-pfx/PuzzleScriptNext/assets/13802168/6490da58-e7f2-4018-8b5a-90e4985e0ac7)
And now this is it, still in PS PLus, with a level only 4 tiles wide
9![image](https://github.com/david-pfx/PuzzleScriptNext/assets/13802168/eca28a7f-a7e9-417c-ab7f-ea4453210ebd)

----

Same level but in PS Next (without my fix)
(notice there are 2 lines between the sections instead of 1)
![image](https://github.com/david-pfx/PuzzleScriptNext/assets/13802168/7ebc28cd-bc72-4725-aa59-91f2c3cf9592)
And now with a level only 4 tiles wide
(The level is so much empty space now, it wasn't there in PS Plus)
![image](https://github.com/david-pfx/PuzzleScriptNext/assets/13802168/683b4725-177f-486f-9d14-86c4fff76b4e)

----

I'll update my PR when I have the solution
